engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -1,44 +1,98 @@
 package cromwell.engine.workflow.lifecycle.deletion
 
-import akka.actor.{LoggingFSM, Props}
+import java.io.FileNotFoundException
+
+import akka.actor.{ActorRef, LoggingFSM, Props}
+import cromwell.core.io._
 import cromwell.core.path.{DefaultPathBuilder, Path, PathBuilder, PathFactory}
-import cromwell.core.{CallOutputs, RootWorkflowId}
+import cromwell.core.{CallOutputs, RootWorkflowId, WorkflowMetadataKeys}
+import cromwell.engine.io.IoAttempts.EnhancedCromwellIoException
 import cromwell.engine.workflow.lifecycle.deletion.DeleteWorkflowFilesActor._
+import cromwell.filesystems.gcs.batch.GcsBatchCommandBuilder
+import cromwell.services.metadata.MetadataService.PutMetadataAction
+import cromwell.services.metadata.impl.FileDeletionStatus
+import cromwell.services.metadata.impl.FileDeletionStatus.{Failed, InProgress, Succeeded}
+import cromwell.services.metadata.{MetadataEvent, MetadataKey, MetadataValue}
 import cromwell.util.GracefulShutdownHelper.ShutdownCommand
-import wom.graph.GraphNodePort.{GraphNodeOutputPort, OutputPort}
+import wom.graph.GraphNodePort.OutputPort
 import wom.types.WomSingleFileType
 import wom.values.{WomArray, WomSingleFile, WomValue}
 
+import scala.concurrent.ExecutionContext
 import scala.util.Try
 
 class DeleteWorkflowFilesActor(rootWorkflowId: RootWorkflowId,
                                workflowFinalOutputs: CallOutputs,
                                workflowAllOutputs: CallOutputs,
-                               pathBuilders: List[PathBuilder]) extends LoggingFSM[DeleteWorkflowFilesActorState, DeleteWorkflowFilesActorStateData] {
+                               pathBuilders: List[PathBuilder],
+                               serviceRegistryActor: ActorRef,
+                               ioActorRef: ActorRef) extends LoggingFSM[DeleteWorkflowFilesActorState, DeleteWorkflowFilesActorStateData] with IoClientHelper {
+
+  implicit val ec: ExecutionContext = context.dispatcher
```

Comment:
ToL: Don't change anything... especially since what you have may be correct for this situation.

I've been reading and watching youtubes on dispatchers. There are some theories / best practice recommendations out there that cpu & i/o should be run on a different thread pool than messaging/events. Via Kaan: https://www.youtube.com/watch?v=g_jP47HFpWA especially around [35:59](https://youtu.be/g_jP47HFpWA?t=2159).

That said, I don't think we need massive performance gains here, but exposing the Actor's messaging-thread-pool as the implicit-thread-pool used by other methods may not be best practice in general.